lib/msal-common/src/cache/CacheManager.ts

```diff
@@ -297,50 +298,92 @@ export abstract class CacheManager implements ICacheManager {
+                return;
             }
 
-            if (!StringUtils.isEmpty(credentialType)) {
-                matches = matches && this.matchCredentialType(entity, credentialType);
+            if (!StringUtils.isEmpty(credentialType) && !this.matchCredentialType(entity, credentialType)) {
+                return;
             }
 
-            if (!StringUtils.isEmpty(clientId)) {
-                matches = matches && this.matchClientId(entity, clientId);
+            if (!StringUtils.isEmpty(clientId) && !this.matchClientId(entity, clientId)) {
+                return;
             }
 
             // idTokens do not have "target", target specific refreshTokens do exist for some types of authentication
             // TODO: Add case for target specific refresh tokens
-            if (!StringUtils.isEmpty(target) && credType === CredentialType.ACCESS_TOKEN) {
-                matches = matches && this.matchTarget(entity, target);
+            if (!StringUtils.isEmpty(target) && !this.matchTarget(entity, target)) {
+                return;
             }
 
-            if (matches) {
-                switch (credType) {
-                    case CredentialType.ID_TOKEN:
-                        matchingCredentials.idTokens[cacheKey] = entity as IdTokenEntity;
-                        break;
-                    case CredentialType.ACCESS_TOKEN:
-                        matchingCredentials.accessTokens[cacheKey] = entity as AccessTokenEntity;
-                        break;
-                    case CredentialType.REFRESH_TOKEN:
-                        matchingCredentials.refreshTokens[cacheKey] = entity as RefreshTokenEntity;
-                        break;
-                }
+            switch (credType) {
+                case CredentialType.ID_TOKEN:
+                    matchingCredentials.idTokens[cacheKey] = entity as IdTokenEntity;
+                    break;
+                case CredentialType.ACCESS_TOKEN:
+                    matchingCredentials.accessTokens[cacheKey] = entity as AccessTokenEntity;
+                    break;
+                case CredentialType.REFRESH_TOKEN:
+                    matchingCredentials.refreshTokens[cacheKey] = entity as RefreshTokenEntity;
+                    break;
             }
         });
 
         return matchingCredentials;
     }
 
+    /**
+     * Removes all app metadata objects from cache.
+     */
+    removeAppMetadata(): boolean {
+        const allCacheKeys = this.getKeys();
+        allCacheKeys.forEach((cacheKey) => {
+            if (this.isAppMetadata(cacheKey)) {
+                this.removeItem(cacheKey, CacheSchemaType.APP_META_DATA);
+            }
+        });
+
+        return true;
+    }
+
+    /**
+     * Removes all accounts and related tokens from cache.
+     */
+    removeAllAccounts(): boolean {
+        const allCacheKeys = this.getKeys();
+        allCacheKeys.forEach((cacheKey) => {
```

Comment:
Instead of repeating logic here, could you just call `getAccountsFilteredByInternal()`? Alternatively, we could factor out the logic to a helper function, something like `isAccount`, that you could call in both places?

Reply:
Done.